src/Symfony/Bridge/Doctrine/ArgumentResolver/EntityValueResolver.php

```diff
@@ -217,8 +220,22 @@ private function findViaExpression(ObjectManager $manager, Request $request, Map
             throw new \LogicException(\sprintf('You cannot use the "%s" if the ExpressionLanguage component is not available. Try running "composer require symfony/expression-language".', __CLASS__));
         }
 
+        $variablesFormInjectors = [];
+        foreach($options->modifiersInjectors ?? $this->defaulInjectors as $injector) {
+            $injectorObject = $this->variablesInjectorLocator->get($injector);
+
+            if (!$injectorObject instanceof EntityValueResolverExpressionModifiersInjectorInterface) {
+                throw new \LogicException(\sprintf('You cannot use "%s" as Expression Modifier, please make it implement "%s".', $injector, EntityValueResolverExpressionModifiersInjectorInterface::class));
+            }
+
+            $variablesFormInjectors = array_merge($variablesFormInjectors, $injectorObject->getVariables());
+            foreach($injectorObject->getFunctions() as $injectorFunction) {
```

Comment:
I am not convinced that we need functions to be added here. In the DoctrineBundle where the `EntityValueResolver` is registered as a service a custom `ExpressionLanguage` is also registered which is used exclusively for the entity value resolver. New functions can be added to this `ExpressionLanguage` instance instead (a new config option in the bundle to make registering them easier might still be an option though).

Reply:
for functions ok but variables? 
it's more easiest to inject an array than a custom function with evaluated and compile callback to do ?

Reply:
see @stof's comments about why using variables is not the best idea: https://github.com/symfony/symfony/pull/57676#discussion_r1668528800

Reply:
For user, I saw it, I extending my PR for have config option and MapEntityAttribute to load it when we really want. 
but insead iser, it can be possible to add env var too.
Yes we can decorate the service, extends evaluate function  and add vars in , but I thought this way of doing things would be simpler and more flexible.